### PR TITLE
Add phone number to free enterprise license request

### LINKED ISSUE #8939: Add phone number field to EnterpriseFreeLicenseForm
## What?
In order to request free enterprise license trials, starting in 4.0 the request needs a phone number filled in.

### PR DESCRIPTION
**Pull request by bernd**
Add phone number to free enterprise license form.

Closes #8939

### CODE REVIEW

File: graylog2-web-interface/src/components/enterprise/EnterpriseFreeLicenseForm.jsx [hunk 1]
```diff
@@ -104,6 +139,15 @@ export default class EnterpriseFreeLicenseForm extends React.Component {
                    placeholder="Please provide a valid email address to send the license key to"
                    required
                    onChange={this.handleInput('email')} />
+            <Input type="text"
+                   id="phone"
+                   label="Phone Number"
+                   value={phone}
+                   placeholder="Please provide a valid phone number"
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/enterprise/EnterpriseFreeLicenseForm.jsx:146]: You ask for a "valid" phone number, and you parse its validness via a custom regex, but we do not explain to the user what a "valid" phone number looks like, nor do we automatically mask the number to render "validly".

File: graylog2-web-interface/src/components/enterprise/EnterpriseFreeLicenseForm.jsx [hunk 2]
```diff
@@ -104,6 +139,15 @@ export default class EnterpriseFreeLicenseForm extends React.Component {
                    placeholder="Please provide a valid email address to send the license key to"
                    required
                    onChange={this.handleInput('email')} />
+            <Input type="text"
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/enterprise/EnterpriseFreeLicenseForm.jsx:142]: How do we feel about using [`tel`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/tel)?

File: graylog2-web-interface/src/components/enterprise/EnterpriseFreeLicenseForm.jsx [hunk 3]
```diff
@@ -5,7 +5,17 @@ import lodash from 'lodash';
 import { Button, ButtonToolbar, Col, Row } from 'components/graylog';
 import { Input } from 'components/bootstrap';
 
-const FORM_FIELDS = ['firstName', 'lastName', 'email', 'company'];
+const FORM_FIELDS = ['firstName', 'lastName', 'email', 'phone', 'company'];
+
+const PHONE_PATTERN = /^[0-9+]+[0-9\-.\s]+$/;
```
Comment (kyleknighted) [inline graylog2-web-interface/src/components/enterprise/EnterpriseFreeLicenseForm.jsx:10]: Should we use HTML5 [`pattern`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefpattern) to validate?

**Comment by bernd** [comment]
@kyleknighted Could you update the PR to fix the issues? :smile: That would be awesome.

**Review by kroepke — APPROVED** [review]
lgtm

**Pull request merged**